### PR TITLE
Remove UTF-8 BOM from changelog command outputs

### LINKED ISSUE #3191: changelog generated yaml utf8 BOM
The YAML generated by docs-builder changelog starts with an UTF-8 BOM (Byte Order Mark)  which may make some tools fail to handle this properly as [reported by Copilot here](https://github.com/elastic/elastic-otel-java/pull/1075#discussion_r3148357396).

I think we should clarify the following:
- is this BOM usage intentional for the generated yaml, for example to help with localization of content or integration with other tooling
- if the potential issue reported by copilot is significant or not in practice, if not, then we should find a way to avoid such noisy comments.

There are two possible approaches depending on the answer to the questions above:
- if it's not intentional nor needed, then we can just remove it as it's simpler
- if it's intentional or required, then we should find a way to make copilot (and others) stop from reporting it as a potential issue.

In practice, the impact of this is minor but not zero because it's mostly noise on the PR reviewer. However it also makes the copilot (or similar) automated review less useful for changelog-related PRs, which keeps some friction in the release process.

### PR DESCRIPTION
**Pull request by lcawl**
Fixes https://github.com/elastic/docs-builder/issues/3191

## AI implementation details

The implementation ensures that:

1. **No existing functionality is changed** - only BOM handling is added
2. **All outputs are normalized** - both fresh generation and file copying operations
3. **Robust testing** - comprehensive unit and integration test coverage
4. **Future-proof** - any BOM introduced by external tools will be stripped

### BOM Detection Utility Created

- **`ChangelogUtf8Normalization`** class in `/src/services/Elastic.Changelog/Utilities/`
- `StripLeadingUtf8BomChar()` - removes BOM character from strings
- `HasUtf8Bom()` - detects BOM in byte arrays
- Proper documentation explaining this is for tooling compatibility

### File Writers Updated

All key YAML writing operations now strip BOMs before writing:
- **`ChangelogFileWriter`** - changelog creation
- **`ChangelogBundlingService`** - bundling operations  
- **`ChangelogBundleAmendService`** - bundle amendments
- **`GitHubReleaseChangelogService`** - GitHub release processing

### File Readers Updated

- **`ChangelogPrepareArtifactService`** - replaced `File.Copy` with read-normalize-write for YAML files
- **`ChangelogCommand`** init - normalizes content when updating existing `changelog.yml`

### Comprehensive Tests Added

- **`ChangelogUtf8NormalizationTests`** - 12 utility tests covering all scenarios
- **`BundleChangelogsTests`** - BOM handling in bundling operations
- **`ChangelogCreationServiceTests`** - BOM-free creation verification
- **`ChangelogPrepareArtifactServiceTests`** - File copying normalization tests

### Verification Complete

- Confirmed existing YAML files in the repo contain no BOMs
- All output paths now guarantee UTF-8 without BOM
- Tests verify byte-level BOM detection and removal

## Generative AI disclosure

1. Did you use a generative AI (GenAI) tool to assist in creating this contribution?
- [x] Yes  
- [ ] No  
2. If you answered "Yes" to the previous question, please specify the tool(s) and model(s) used (e.g., Google Gemini, OpenAI ChatGPT-4, etc.).

Tool(s) and model(s) used: composer-2, claude-4-sonnet-thinking

### CODE REVIEW

File: src/services/Elastic.Changelog/Bundling/ChangelogBundleAmendService.cs [hunk 1]
```diff
@@ -256,7 +257,9 @@ public async Task<bool> AmendBundle(IDiagnosticsCollector collector, AmendBundle
 			if (!string.IsNullOrWhiteSpace(outputDir) && !_fileSystem.Directory.Exists(outputDir))
 				_ = _fileSystem.Directory.CreateDirectory(outputDir);
 
-			await _fileSystem.File.WriteAllTextAsync(amendFilePath, yaml, Encoding.UTF8, ctx);
+			// Strip any leading BOM to ensure clean UTF-8 output for tooling compatibility
+			var normalizedYaml = ChangelogUtf8Normalization.StripLeadingUtf8BomChar(yaml);
+			await _fileSystem.File.WriteAllBytesAsync(amendFilePath, Encoding.UTF8.GetBytes(normalizedYaml), ctx);
```
Comment (Mpdreamz) [inline src/services/Elastic.Changelog/Bundling/ChangelogBundleAmendService.cs]: I believe we can simply do `WriteAllText` with `new UTF8Encoding(false)` instead of `Encoding.UTF8`. 

That way we allocate less strings, see also: https://danielwertheim.se/utf-8-bom-adventures-in-c/

**Review by Mpdreamz — CHANGES_REQUESTED** [review]

File: src/services/Elastic.Changelog/Bundling/ChangelogBundleAmendService.cs (hunk 1 above)
Comment (lcawl) [reply src/services/Elastic.Changelog/Bundling/ChangelogBundleAmendService.cs]: 460f419e0cb6ff8af88b4615d40784e533bb7731  is my attempt to address this comment as well as https://github.com/elastic/docs-builder/pull/3243#pullrequestreview-4221468597

The AI summary for that change is as follows:

## Summary of Changes in 460f419e0cb6ff8af88b4615d40784e533bb7731

This commit addresses PR review feedback by **consolidating BOM helper duplication** and **optimizing UTF-8 write style**:

### 🔄 **Consolidated BOM/String Normalization**

- **Created shared utility**: New `Utf8TextNormalization.cs` in `Elastic.Documentation/Text/` with enhanced BOM stripping that removes **all consecutive** leading U+FEFF characters (handles double-BOM scenarios)
- **Updated `ReleaseNotesSerialization.cs`**: Removed private `StripLeadingUtf8BomChar` method, now uses shared `Utf8TextNormalization.StripLeadingUtf8Bom`
- **Converted `ChangelogUtf8Normalization.cs`**: Now serves as a thin forwarder to the shared implementation, maintaining backward compatibility

### ⚡ **Optimized UTF-8 File Writes**

Replaced `WriteAllBytesAsync(path, Encoding.UTF8.GetBytes(content))` with `WriteAllTextAsync(path, content, Utf8NoBom)` in all changelog writers:
- `ChangelogFileWriter.cs`
- `ChangelogBundlingService.cs`
- `ChangelogBundleAmendService.cs`
- `GitHubReleaseChangelogService.cs`
- `ChangelogPrepareArtifactService.cs`

Each now uses `static readonly UTF8Encoding Utf8NoBom = new(encoderShouldEmitUTF8Identifier: false)` to avoid per-write byte array allocations.

### ✅ **Enhanced Testing**

- **New test file**: `Utf8TextNormalizationTests.cs` with comprehensive shared utility tests
- **Enhanced existing tests**: Added consecutive BOM test cases in `ChangelogUtf8NormalizationTests.cs`
- **Edge case coverage**: Tests for multiple BOMs, BOM-only strings, and BOMs in middle/end positions

The changes eliminate code duplication, improve memory efficiency, and provide more robust BOM handling across the codebase.

**Review by Mpdreamz — APPROVED** [review]
